app/models/activity.rb

```diff
@@ -58,6 +58,25 @@ def generate_breadcrumbs
     generate_breadcrumb subject if subject
   end
 
+  def self.url_search_query(filters)
```

Comment:
This is construction of URL GET query. Isn't this always gonna be used on the front-end? In that case, I just think this would be better off being located in some `helper`, e.g. `activity_helper`. What do you think?

Reply:
I think we can use anywhere, not only in front end, but also in back end for redirected links. And you don't need to include it everytime.